src/tape/ensemble.py

```diff
@@ -1094,6 +1094,59 @@ def from_source_dict(self, source_dict, column_mapper=None, npartitions=1, **kwa
         self._object_dirty = False
         return self
 
+    def convert_flux_to_mag(self, flux_col, zero_point, err_col=None, zp_form="mag", out_col_name=None):
+        """Converts a flux column into a magnitude.
+
+        Parameters
+        ----------
+        flux_col: 'str'
+            The name of the ensemble flux column to convert into magnitudes.
+        zero_point: 'str'
+            The name of the ensemble column containing the zero point
+            information for column transformation.
+        err_col: 'str', optional
+            The name of the ensemble column containing the errors to propagate.
+        zp_form: `str`
+            The form of the zero point column, either "flux" or
+            "magnitude"/"mag". Determines how the zero point (zp) is applied in
+            the conversion. If "flux", then the function is applied as
+            Mag=-2.5*log10(flux/zp), or if "magnitude", then
+            Mag=-2.5*log10(flux)+zp.
+        out_col_name: 'str', optional
+            The name of the output magnitude column, if None then the output
+            is just the flux column name + "_mag". The error column is also
+            generated as the out_col_name + "_err".
+
+        Returns
+        ----------
+        ensemble: `tape.ensemble.Ensemble`
+            The ensemble object with a new mag column.
+
+        """
+        if out_col_name is None:
+            out_col_name = flux_col + "_mag"
+
+        if zp_form == "flux":  # mag = -2.5*np.log10(flux/zp)
+            self._source = self._source.assign(
+                **{out_col_name: lambda x: -2.5 * np.log10(x[flux_col] / x[zero_point])}
+            )
+
+        elif zp_form == "magnitude" or zp_form == "mag":  # mag = -2.5*np.log10(flux) + zp
+            self._source = self._source.assign(
+                **{out_col_name: lambda x: -2.5 * np.log10(x[flux_col]) + x[zero_point]}
+            )
+
+        else:
+            raise ValueError(f"{zp_form} is not a valid zero_point format.")
+
+        # Calculate Errors
+        if err_col is not None:
+            self._source = self._source.assign(
+                **{out_col_name + "_err": lambda x: (2.5 / np.log(10)) * (x[err_col] / x[flux_col])}
```

Comment:
I would add a note or a comment in the docstring to note that this is only correct if the limit where the error on the flux is much smaller than the flux itself. This is not an exact formula, and that should be noted somewhere.

Reply:
Will do, is there a preferable way to calculate errors here?

Reply:
I _think_ it is fine. Symmetric errors in flux create nonsymettric errors in log, i.e., one would have to create different upper and lower errors. For most of LSST like purposes I expect errors to be much smaller than the flux and the equation is ok I do not think we want to go beyond this implementation at this point.